<a href="https://colab.research.google.com/github/PurpleState/neural-tts/blob/main/01_first_tts_inference.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Neural TTS

## Objective

Generate speech from text using a pretrained TTS model.

This notebook helps me:

- Verify environment setup
- Understand the TTS pipeline
- Generate my first speech sample
- Prepare for FastPitch fine-tuning later

Target Architecture:

Text
→ FastPitch
→ Mel Spectrogram
→ HiFiGAN
→ Speech

In [4]:
import sys

print(sys.version)

3.12.13 (main, Mar  4 2026, 09:23:07) [GCC 11.4.0]


In [5]:
import torch

print(torch.__version__)
print(torch.cuda.is_available())

2.11.0+cpu
False


### Using `gTTS` (Google Text-to-Speech) as an alternative

Since `TTS` does not support Python 3.12, we can use `gTTS` as an alternative. `gTTS` is a Python library and CLI tool to interface with Google Translate's text-to-speech API.

In [9]:
pip install gTTS

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.2/98.2 kB 6.1 MB/s eta 0:00:00
  Attempting uninstall: click
    Found existing installation: click 8.4.1
    Uninstalling click-8.4.1:
      Successfully uninstalled click-8.4.1
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
huggingface-hub 1.17.0 requires click>=8.4.0, but you have click 8.1.8 which is incompatible.
typer 0.25.1 requires click>=8.2.1, but you have click 8.1.8 which is incompatible.
wandb 0.27.0 requires click>=8.2.0, but you have click 8.1.8 which is incompatible.


In [10]:
from gtts import gTTS
from IPython.display import Audio

text = "Hello, this is a test of the Google Text-to-Speech library."
tts = gTTS(text=text, lang='en')
tts.save("output.mp3")

Audio("output.mp3")

In [12]:
from gtts.lang import tts_langs

# Get a dictionary of supported languages
supported_languages = tts_langs()

print("Supported languages for gTTS:")
for lang_code, lang_name in supported_languages.items():
    print(f"  {lang_code}: {lang_name}")

Supported languages for gTTS:
  af: Afrikaans
  am: Amharic
  ar: Arabic
  bg: Bulgarian
  bn: Bengali
  bs: Bosnian
  ca: Catalan
  cs: Czech
  cy: Welsh
  da: Danish
  de: German
  el: Greek
  en: English
  es: Spanish
  et: Estonian
  eu: Basque
  fi: Finnish
  fr: French
  fr-CA: French (Canada)
  gl: Galician
  gu: Gujarati
  ha: Hausa
  hi: Hindi
  hr: Croatian
  hu: Hungarian
  id: Indonesian
  is: Icelandic
  it: Italian
  iw: Hebrew
  ja: Japanese
  jw: Javanese
  km: Khmer
  kn: Kannada
  ko: Korean
  la: Latin
  lt: Lithuanian
  lv: Latvian
  ml: Malayalam
  mr: Marathi
  ms: Malay
  my: Myanmar (Burmese)
  ne: Nepali
  nl: Dutch
  no: Norwegian
  pa: Punjabi (Gurmukhi)
  pl: Polish
  pt: Portuguese (Brazil)
  pt-PT: Portuguese (Portugal)
  ro: Romanian
  ru: Russian
  si: Sinhala
  sk: Slovak
  sq: Albanian
  sr: Serbian
  su: Sundanese
  sv: Swedish
  sw: Swahili
  ta: Tamil
  te: Telugu
  th: Thai
  tl: Filipino
  tr: Turkish
  uk: Ukrainian
  ur: Urdu
  vi: Vietnamese
  